pysmt/numeral.py

```diff
@@ -0,0 +1,21 @@
+import warnings
+
+use_z3 = False
+try:
+    import z3num
+    use_z3 = True
+except:
+    pass
+
+if use_z3:
+    class Numeral(z3num.Numeral):
+        """Represents a Number (Algebraic)"""
+        def __hash__(self):
+            return hash(self.sexpr())
+
+else:
+    class Numeral(object):
+        """Represents a Number (Algebraic)"""
+        def __init__(self, obj):
+            self.obj = obj
+            warnings.warn("Z3 is not installed. We currently do not support stand-alone algebraic numbers. This might lead to unexpected results")
```

Comment:
I agree that this should never happen with the code in this PR, so I think an exception is better to signal future issues on this matter.